google/pubsub/v1/pubsub_gapic.yaml

```diff
@@ -221,6 +240,71 @@ interfaces:
     field_name_patterns:
       subscription: subscription
     timeout_millis: 60000
+  - name: ListSnapshots
+    flattening:
+      groups:
+      - parameters:
+        - project
+    required_fields:
+    - project
+    request_object_method: true
+    page_streaming:
+      request:
+        page_size_field: page_size
+        token_field: page_token
+      response:
+        token_field: next_page_token
+        resources_field: snapshots
+    retry_codes_name: idempotent
+    retry_params_name: default
+    field_name_patterns:
+      project: project
+    timeout_millis: 60000
+  - name: CreateSnapshot
+    flattening:
+      groups:
+      - parameters:
+        - name
+        - subscription
+    required_fields:
+    - subscription
+    request_object_method: true
+    retry_codes_name: idempotent
+    retry_params_name: default
+    field_name_patterns:
+      name: snapshot
+      subscription: subscription
+    timeout_millis: 60000
+  - name: DeleteSnapshot
+    flattening:
+      groups:
+      - parameters:
+        - snapshot
+    required_fields:
+    - snapshot
+    request_object_method: false
+    retry_codes_name: idempotent
+    retry_params_name: default
+    field_name_patterns:
+      snapshot: snapshot
+    timeout_millis: 60000
+  - name: Seek
+    flattening:
+      groups:
+      - parameters:
+        - subscription
+        - time
+      - parameters:
+        - subscription
+        - snapshot
```

Comment:
Yes, two messages will be generated. This isn't robust though, because if a third variant of the oneof is added which is of type Time or String, then we can't generate an overload for it. I think we need support for OneOf flattened parameters, as in https://github.com/googleapis/toolkit/issues/1057 . For the short term, just don't do any flattening at all.